### ISSUE TITLE
no_yoda_conditions for variables

### ISSUE THREAD

**Issue by DGRHicron** [abaplint/abaplint#3539]
```
REPORT zfoo.

START-OF-SELECTION.
  DATA:
     BEGIN OF stru,
      is_valid TYPE abap_bool,
      END OF stru.
  DATA(error_flag) = xsdbool( abap_false = stru-is_valid ).
  WRITE error_flag.
```

IMHO, that part of code should trigger no_yoda_conditions issue.
